### PR TITLE
ocsinventory-agent: init at 2.10.1

### PR DESCRIPTION
**Pull request by anthonyroussel**
## Description of changes

The purpose of this PR is to continue the work started by @totoroot on PR https://github.com/NixOS/nixpkgs/pull/242659, to make the OCS Inventory agent work on NixOS environment.

I have seen this tool several times in work environments in French companies. This tool was each time a prerequisite for developers to be able to install a Linux OS on their workstation, since IT and system administrators need to trace and inventory the hardware/software.  I think it might be a good idea to add support for this package on NixOS.

I'll continue after this PR on the development of the NixOS module.

See https://github.com/OCSInventory-NG/UnixAgent

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by totoroot** [comment]
Result of `nixpkgs-review pr 264172` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>18 packages built:</summary>
  <ul>
    <li>ocsinventory-agent</li>
    <li>ocsinventory-agent.devdoc</li>
    <li>perl536Packages.NetCUPS</li>
    <li>perl536Packages.NetCUPS.devdoc</li>
    <li>perl536Packages.ParseEDID</li>
    <li>perl536Packages.ParseEDID.devdoc</li>
    <li>perl536Packages.ProcDaemon</li>
    <li>perl536Packages.ProcDaemon.devdoc</li>
    <li>perl536Packages.ProcPIDFile</li>
    <li>perl536Packages.ProcPIDFile.devdoc</li>
    <li>perl538Packages.NetCUPS</li>
    <li>perl538Packages.NetCUPS.devdoc</li>
    <li>perl538Packages.ParseEDID</li>
    <li>perl538Packages.ParseEDID.devdoc</li>
    <li>perl538Packages.ProcDaemon</li>
    <li>perl538Packages.ProcDaemon.devdoc</li>
    <li>perl538Packages.ProcPIDFile</li>
    <li>perl538Packages.ProcPIDFile.devdoc</li>
  </ul>
</details>

**Review by totoroot — APPROVED** [review]
Thank you so much for working on this! It builds fine for me and seems to work as expected.

```sh
> sudo ./results/ocsinventory-agent/bin/ocsinventory-agent --help

Usage:
	--backend-collect-timeout set a max delay time of one action (search package id, ...) is set (600)
	--basevardir=/path  indicate the directory where should the agent store its files (/var/lib/ocsinventory-agent)
	-d  --daemon        detach the agent in background (0)
	   --debug         debug mode (0)
	   --devlib        search for Backend mod in ./lib only (0)
	-f --force          always send data to server (Don't ask before) (0)
	-i --info           verbose mode (1)
	--lazy              do not contact the server more than one time during the PROLOG_FREQ (0)
	-l --local=DIR      do not contact server but write inventory in DIR directory in XML ()
	  --logfile=FILE   log message in FILE ()
	-p --password=PWD   password for server auth
	-P --proxy=PROXY    proxy address. e.g: http://user:pass@proxy:port ()
	-r --realm=REALM    realm for server auth. e.g: 'Restricted Area' ()
	-s --server=uri     server uri (http://ocsinventory-ng/ocsinventory)
	  --stdout         do not write or post the inventory but print it on STDOUT
	-t --tag=TAG        use TAG as tag (). Will be ignored by server if a value already exists.
	-u --user=USER      user for server auth ()
	  --version        print the version
	-w --wait=seconds   wait a random period before contacting server like --daemon does ()
	--nosoftware        do not return installed software list (0)
	--delaytime	     set a max delay time (in second) if no PROLOG_FREQ is set (3600)
	--scan-homedirs     permit to scan home user directories (0)
	--ssl=0|1           disable or enable SSL communications check (1)
	--ca=FILE           path to CA certificates file in PEM format ()

Manpage:
	See man ocsinventory-agent

Ocsinventory-Agent is released under GNU GPL 2 license
```

File: pkgs/by-name/oc/ocsinventory-agent/package.nix [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      to help a network or system administrator to keep track of the hardware and
+      software configurations of computers that are installed on the network.
+    '';
+    homepage = "https://ocsinventory-ng.org";
+    changelog = "https://github.com/OCSInventory-NG/UnixAgent/releases/tag/v${version}";
+    downloadPage = "https://github.com/OCSInventory-NG/UnixAgent/releases";
+    license = licenses.gpl2;
+    mainProgram = "ocsinventory-agent";
+    maintainers = with maintainers; [ totoroot anthonyroussel ];
+    platforms = platforms.linux;
```
Comment (totoroot) [inline pkgs/by-name/oc/ocsinventory-agent/package.nix:93]: We should try to support macOS as well, but IMO we can target this later on. [Version 2.10.1](https://github.com/OCSInventory-NG/UnixAgent/releases/tag/v2.10.1-MAC) should work fine on macOS as well, but for me, it's okay to get it out the door for Linux first.

**Review by totoroot — APPROVED** [review]

File: pkgs/by-name/oc/ocsinventory-agent/package.nix (hunk 1 above)
Comment (anthonyroussel) [reply pkgs/by-name/oc/ocsinventory-agent/package.nix:93]: Yes we should add Darwin support for this package.

However I don't own any macOS workstations to test the `ocsinventory-agent` package, and to create the `Mac::SysProfile` perl dependency package.

Comment (Janik-Haag) [reply pkgs/by-name/oc/ocsinventory-agent/package.nix:93]: You could ask on the

**Comment by anthonyroussel** [comment]
@ofborg build ocsinventory-agent

**Comment by natsukium** [comment]
@ofborg eval

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -14657,6 +14657,36 @@ with self; {
...
+    version = "1.504";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/B/BD/BDFOY/Mac-PropertyList-1.504.tar.gz";
+      hash = "sha256-aIl96Yw2j76c22iF1H3qADxG7Ho3MmNSPvZkVwc7eq4=";
+    };
+    propagatedBuildInputs = [ XMLEntities ];
+    meta = {
+      description = "Work with Mac plists at a low level";
+      homepage = "https://github.com/briandfoy/mac-propertylist";
+      license = with lib.licenses; [ artistic1 gpl1Plus ];
```
Comment (anthonyroussel) [inline pkgs/top-level/perl-packages.nix:14671]:
```suggestion
      license = lib.licenses.artistic2;
```

File: pkgs/top-level/perl-packages.nix [hunk 3]
```diff
@@ -19665,6 +19711,20 @@ with self; {
...
+    pname = "Parse-Edid";
+    version = "1.0.7";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/G/GR/GROUSSE/Parse-EDID-1.0.7.tar.gz";
+      hash = "sha256-GtwPEFoyGYoqK02lsOD5hfBe/tmc42YZCnkOFl1nW/E=";
+    };
+    buildInputs = [ TestWarn ];
+    meta = {
+      description = "Extended display identification data (EDID) parser";
+      license = with lib.licenses; [ gpl3Only ];
```
Comment (anthonyroussel) [inline pkgs/top-level/perl-packages.nix:19724]:
```suggestion
      license = lib.licenses.gpl3Plus
```

File: pkgs/by-name/oc/ocsinventory-agent/package.nix (hunk 1 above)
Comment (anthonyroussel) [reply pkgs/by-name/oc/ocsinventory-agent/package.nix]: I managed to get a green build for x86_64-darwin from Ofborg.
I'll ask volunteers on #macos:nixos.org to check that the binary works :)

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 264172` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>perl536Packages.NetCUPS</li>
    <li>perl536Packages.NetCUPS.devdoc</li>
    <li>perl538Packages.NetCUPS</li>
    <li>perl538Packages.NetCUPS.devdoc</li>
  </ul>
</details>
<details>
  <summary>26 packages built:</summary>
  <ul>
    <li>ocsinventory-agent</li>
    <li>ocsinventory-agent.devdoc</li>
    <li>perl536Packages.MacPropertyList</li>
    <li>perl536Packages.MacPropertyList.devdoc</li>
    <li>perl536Packages.MacSysProfile</li>
    <li>perl536Packages.MacSysProfile.devdoc</li>
    <li>perl536Packages.ParseEDID</li>
    <li>perl536Packages.ParseEDID.devdoc</li>
    <li>perl536Packages.ProcDaemon</li>
    <li>perl536Packages.ProcDaemon.devdoc</li>
    <li>perl536Packages.ProcPIDFile</li>
    <li>perl536Packages.ProcPIDFile.devdoc</li>
    <li>perl536Packages.XMLEntities</li>
    <li>perl536Packages.XMLEntities.devdoc</li>
    <li>perl538Packages.MacPropertyList</li>
    <li>perl538Packages.MacPropertyList.devdoc</li>
    <li>perl538Packages.MacSysProfile</li>
    <li>perl538Packages.MacSysProfile.devdoc</li>
    <li>perl538Packages.ParseEDID</li>
    <li>perl538Packages.ParseEDID.devdoc</li>
    <li>perl538Packages.ProcDaemon</li>
    <li>perl538Packages.ProcDaemon.devdoc</li>
    <li>perl538Packages.ProcPIDFile</li>
    <li>perl538Packages.ProcPIDFile.devdoc</li>
    <li>perl538Packages.XMLEntities</li>
    <li>perl538Packages.XMLEntities.devdoc</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

File: pkgs/by-name/oc/ocsinventory-agent/package.nix (hunk 1 above)
Comment (anthonyroussel) [reply pkgs/by-name/oc/ocsinventory-agent/package.nix]: @wegank Thanks for your help! :)

**Review by Janik-Haag — APPROVED** [review]
I'm not familiar with perl packaging but in the diff looks good and the package seems to work fine.

**Comment by Endle** [comment]
I testes it on my macOS (14.0) with M2. Log attached

```
nix build .#ocsinventory-agent --extra-experimental-features nix-command --extra-experimental-features flakes
path '/Volumes/extend/src/nixpkgs/pkgs/by-name/oc/ocsinventory-agent' does not contain a 'flake.nix', searching up
```


```
$ ./result/bin/ocsinventory-agent --local=/tmp --debug
```
log link: https://gist.github.com/Endle/b236ab0b3e66a62dcf3f79638c506de2

**Comment by anthonyroussel** [comment]
PR for the NixOS module of the OCS Inventory agent: https://github.com/NixOS/nixpkgs/pull/264910

**Pull request merged**